inspirehep/modules/records/receivers.py

```diff
@@ -106,3 +114,24 @@ def match_valid_experiments(recid, json, *args, **kwargs):
                     # If the experiment cannot be matched it is considered valid
                     facet_experiment = exp.get("experiment")
                 exp.update({"facet_experiment": facet_experiment})
+
+
+@before_record_insert.connect
+@before_record_update.connect
+def dois_clean(sender, *args, **kwargs):
+    dois = []
+    new_l = []
+    if 'dois' in sender:
+        for element in sender['dois']:
+            if 'value' in element:
+                dois.append(element['value'])
+
+        if len(list(set(dois))) > 1:
+            for element in sender['dois']:
```

Comment:
The thing is that we have some dois that are currently like this:
[{u'source': u'bibmatch', u'value': u'10.1007/JHEP04(2015)124'}, {u'value': u'10.1007/JHEP04(2015)124'}]
where they should be considered as duplicated. So what I tried to do is to put only the values in the dois list and then  by this line len(list(set(dois))) if the list contains more than one means that sender['dois'] has multiple values, otherwise has duplicated so we keep only one the first. I did it so as to keep also the source field, so in the end to have sth like this:
[{u'source': u'bibmatch', u'value': u'10.1007/JHEP04(2015)124'}]
Makes sense?